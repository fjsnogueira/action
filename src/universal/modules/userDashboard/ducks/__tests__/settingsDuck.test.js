import reducer, {
  ACTIVITY_WELCOME,
  setActivity,
  setWelcomeActivity,
  clearActivity
} from '../settingsDuck';

test('initial state', () => {
  const initialState = reducer();
  expect(initialState).toMatchSnapshot();
});

test('setActivity() ACTIVITY_WELCOME updates activity, nextPage', () => {
  expect(
    reducer(undefined,
      setActivity(ACTIVITY_WELCOME, '/team/baddad')
    )
  ).toMatchSnapshot();
});

test('setActivity() of invalid activity throws error', () => {
  expect(() => setActivity('narf!', '/team/d00d00')).toThrow();
});

test('setWelcomeActivity() updates activity, nextPage', () => {
  const state = reducer();

  expect(
    reducer(state,
      setWelcomeActivity('/team/baddad')
    )
  ).toMatchSnapshot();
});

test('clearWelcomeActivity() resets state', () => {
  const state = reducer();

  expect(
    reducer(state, clearActivity())
  ).toMatchSnapshot();
});
