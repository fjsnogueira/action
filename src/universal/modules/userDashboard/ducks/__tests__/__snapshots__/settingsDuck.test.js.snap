// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clearWelcomeActivity() resets state 1`] = `
Object {
  "activity": null,
  "nextPage": null,
}
`;

exports[`initial state 1`] = `
Object {
  "activity": null,
  "nextPage": null,
}
`;

exports[`setActivity() ACTIVITY_WELCOME updates activity, nextPage 1`] = `
Object {
  "activity": "welcome",
  "nextPage": "/team/baddad",
}
`;

exports[`setWelcomeActivity() updates activity, nextPage 1`] = `
Object {
  "activity": "welcome",
  "nextPage": "/team/baddad",
}
`;
